packages/cli/src/controllers/oauth/abstract-oauth.controller.ts

```diff
@@ -6,24 +6,35 @@ import type { ICredentialDataDecryptedObject, IWorkflowExecuteAdditionalData } f
 import { jsonParse, ApplicationError } from 'n8n-workflow';
 import { Service } from 'typedi';
 
-import { RESPONSE_ERROR_MESSAGES } from '@/constants';
+import { RESPONSE_ERROR_MESSAGES, Time } from '@/constants';
 import { CredentialsHelper } from '@/credentials-helper';
 import type { CredentialsEntity } from '@/databases/entities/credentials-entity';
 import { CredentialsRepository } from '@/databases/repositories/credentials.repository';
 import { SharedCredentialsRepository } from '@/databases/repositories/shared-credentials.repository';
+import { AuthError } from '@/errors/response-errors/auth.error';
 import { BadRequestError } from '@/errors/response-errors/bad-request.error';
 import { NotFoundError } from '@/errors/response-errors/not-found.error';
 import { ExternalHooks } from '@/external-hooks';
 import type { ICredentialsDb } from '@/interfaces';
 import { Logger } from '@/logging/logger.service';
-import type { OAuthRequest } from '@/requests';
+import type { AuthenticatedRequest, OAuthRequest } from '@/requests';
 import { UrlService } from '@/services/url.service';
 import * as WorkflowExecuteAdditionalData from '@/workflow-execute-additional-data';
 
-export interface CsrfStateParam {
+type CsrfStateParam = {
+	/** Id of the oAuth credential in the DB */
 	cid: string;
+	/** Random CSRF token, used to verify the signature of the CSRF state */
 	token: string;
-}
+	/** Creation timestamp of the CSRF state. Used for expiration.  */
+	createdAt: number;
+	/** User who initiated OAuth flow, included to prevent cross-user credential hijacking. Optional only if `skipAuthOnOAuthCallback` is enabled. */
+	userId?: string;
+};
+
+// TODO: Flip this flag in v2
+// https://linear.app/n8n/issue/CAT-329
+export const skipAuthOnOAuthCallback = process.env.N8N_SKIP_AUTH_ON_OAUTH_CALLBACK !== 'true';
```

Comment:
we don't want to document this because we really don't want anyone using it unless they have no other option, like some embed customers.
Once we have those customers migrated, we'll just remove this, and make auth always mandatory on callback.